### PR TITLE
[ADD][10.0] Add partner_company_type module

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: partner_business_entity/README.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/134/issues>`_. In case of trouble, please
```
Comment (adrienpeiffer) [inline partner_business_entity/README.rst]: Bad URL

**Review by adrienpeiffer — APPROVED** [review]
Small comment/question. Otherwise :+1:

File: partner_business_entity/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    'name': 'Partner Business Entity',
+    'summary': """
+        Adds a business entity to partner that are companies""",
+    'version': '10.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu',
+    'depends': ['base',
+                'sales_team'
```
Comment (adrienpeiffer) [inline partner_business_entity/__manifest__.py]: Why a dependency on sales_team ?

Comment (rousseldenis) [reply partner_business_entity/__manifest__.py]: Yes, that question deserves debate...

Has anyone an advice on this as Odoo split contact action on 'base' module and menu item on 'sales_team' one ?

Comment (sbidoul) [reply partner_business_entity/__manifest__.py]: @rousseldenis What is the debate about?

**Comment by sbidoul** [comment]
It took me some time to understand what this is about. I'd rather name this module `partner_company_type` what do you think?

**Comment by rousseldenis** [comment]
@sbidoul  Yes, this could be more explicit. @adrienpeiffer what do you think as you approved it yet?

**Comment by adrienpeiffer** [comment]
As standard menus are definied in sales_team menu, you don't really have the choice ...

**Comment by lmignon** [comment]
I've the same problem with partner_identification in V10.We are bothered by a 'Sale' menu only to get access to the configurations  menu of contacts
image

**Comment by rousseldenis** [comment]
Maybe is it the time to ask problem to the community and add module to get Configuration Menu in 'Contacts' module ?

**Review by antoniocanovas — APPROVED** [review]
Ok funcionality

**Comment by rafaelbn** [comment]
Hi @rousseldenis thanks for this contribution. About https://github.com/OCA/partner-contact/pull/376#issuecomment-275074800 this functionality must not depends in Contact IMHO. As it is in this moment it's OK.

My only question is: why don't you maintains same field names as in v8? (as you are recovering a functionality)

**Comment by rousseldenis** [comment]
@rafaelbn The 'title' field seemed to me related more to people than companies. That's why I did it in that way (having in mind that we can make the same for people in another module).

File: partner_company_type/README.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Usage
+=====
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/134/10.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (lasley) [inline partner_company_type/README.rst]: Remove comment

File: partner_company_type/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Partner Company Type',
+    'summary': """
+        Adds a company type to partner that are companies""",
```
Comment (lasley) [inline partner_company_type/__manifest__.py]: This is adding blank whitespace. Please use single line strings, optionally concatenated if too long:

```
'summary': 'my long'
           'summary',
```

File: partner_company_type/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    'version': '10.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu',
+    'depends': ['base',
+                'sales_team'
```
Comment (lasley) [inline partner_company_type/__manifest__.py]: Trailing comma

File: partner_company_type/models/res_partner_company_type.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+class ResPartnerCompanyType(models.Model):
+
+    _name = 'res.partner.company.type'
+    _description = 'Partner Company Type'
+
+    name = fields.Char(string='Title', required=True, translate=True)
+    shortcut = fields.Char(string='Abbreviation', translate=True)
+
+    _sql_constraints = [('name_uniq', 'unique (name)',
+                         "Partner Company Type already exists !")]
```
Comment (lasley) [inline partner_company_type/models/res_partner_company_type.py]: No need for a space between `exists` and `!`

File: partner_company_type/views/res_partner.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+	    <field name="model">res.partner</field>
+	    <field name="inherit_id" ref="base.view_partner_form"/>
+	    <field name="arch" type="xml">
+	    	<field name="title" position="after">
+	    		<field name="company_type_id" options='{"no_open": True}'
+	    		attrs="{'invisible': [('is_company', '=', False)]}"/>
+	    	</field>
+	   	</field>
+	</record>
+</odoo>
```
Comment (lasley) [inline partner_company_type/views/res_partner.xml]: Add a newline at the end of file please

**Review by lasley — APPROVED** [review]
Thanks @rousseldenis - few minor comments; none functionally related

**Comment by rousseldenis** [comment]
@rafaelbn In fact, the field 'title' exists in base module, but if you look at res.partner form, Odoo hides it for companies, so it confirms what I've said here above, title is for partners that are people.

**Comment by rousseldenis** [comment]
@lasley Ok, I will solve them.

Moreover, I will change the name of the field in res.partner as it may cause translation problems (Ask me if you want more explanation).

**Comment by rousseldenis** [comment]
Commits are squashed, please let these ones

**Comment by rafaelbn** [comment]
OK, thanks for explanations.

**Comment by lasley** [comment]
> Moreover, I will change the name of the field in res.partner as it may cause translation problems (Ask me if you want more explanation).

I'm curious, why?

**Comment by rousseldenis** [comment]
@lasley That's due to how the translation parser is implemented and due to the model and field are named.

In this case, we had a model 'res.partner.company.type' and a field 'company_type_id' on 'res.partner'.

So, the translation in po file could be ambiguous as there is a field 'id' on the related model. The content of the po file wille be 'res_partner_company_type_id' for that 'id' field.

But, the field company_type_id should have been also 'res_partner_company_type_id' in po.

Odoo will add a trailing id (the id of the ir model field) in po and become res_partner_company_type_id_xxx.

So, for projects it cannot be 'generic' and dependent on the DB you are.

I hope I fed your curiosity ;-)

**Comment by lasley** [comment]
Oh wow that's an interesting edge case - thank you for explaining @rousseldenis!

**Pull request merged**